cmd/crypto/error.go

```diff
@@ -14,60 +14,72 @@
 
 package crypto
 
-import "errors"
+import (
+	"fmt"
+)
 
 // Error is the generic type for any error happening during decrypting
 // an object. It indicates that the object itself or its metadata was
 // modified accidentally or maliciously.
-type Error string
+type Error struct {
+	Err error
+}
+
+// Errorf - formats according to a format specifier and returns
+// the string as a value that satisfies error of type crypto.Error
+func Errorf(format string, a ...interface{}) error {
+	return Error{Err: fmt.Errorf(format, a...)}
+}
 
-func (e Error) Error() string { return string(e) }
+func (e Error) Error() string {
```

Comment:
```suggestion
func (e Error) Error() string { return string(e) }
```

There is no need to change the current type since the we don't add any additional context.
Further, we can keep error constants (instead of vars). In any case if we would need to make `Error` a struct on its own, the it should have `Err` as a private member - e.g. `type Error struct { cause error // ... more fields }` and should implement `func (e *Error) Unwrap() error { return e.cause }`. See: https://blog.golang.org/go1.13-errors

Reply:
I would implement that, I don't like the current string type and also it makes it less context-aware verbatim strings. Errors are context-based we have dynamic errors being generated they should be dynamic as a whole. 

The entire cmd/crypto should generate a proper typed error such that we can send a proper error response, so I would suggest that we move away from the `string` style instead make it dynamic.

Reply:
> I would implement that, I don't like the current string type and also it makes it less context-aware verbatim strings. Errors are context-based we have dynamic errors being generated they should be dynamic as a whole.

I don't have a strong option about this. So, if you and @donatello are fine with this change, I'm okay as well. I just want to point out that a changing the type to a `struct` is not necessary for this change **and** that `type Error string` is in fact a proper type - it's not a string. In particularly, it's not a type-alias (that would be `type Error = string`).
As long as we don't need additional context fields (e.g. a time, error code or something associated with the error value) a primitive type is fine - and allows error constants.

Reply:
its a first step @aead we may have more, so it is better to implement the Unwrap which I just did and proceed.

Reply:
As said, that can be done when necessary. However, if you want to do it now we can.